client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx

```diff
@@ -57,70 +57,105 @@ import {
+import type { OnChangeItemVariant } from '../components/item-variation-picker';
 
 const debug = debugFactory( 'calypso:composite-checkout:wp-checkout' );
 
-const ContactFormTitle = () => {
+// This will make converting to TS less noisy. The order of components can be reorganized later
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+const ContactFormTitle = (): JSX.Element => {
 	const translate = useTranslate();
 	const isActive = useIsStepActive();
 	const isComplete = useIsStepComplete();
 	const { responseCart } = useShoppingCart();
 	const contactDetailsType = getContactDetailsType( responseCart );
 
 	if ( contactDetailsType === 'domain' ) {
-		return ! isActive && isComplete
-			? translate( 'Contact information' )
-			: translate( 'Enter your contact information' );
+		return (
+			<>
+				{ ! isActive && isComplete
+					? String( translate( 'Contact information' ) )
+					: String( translate( 'Enter your contact information' ) ) }
+			</>
+		);
 	}
 
 	if ( contactDetailsType === 'gsuite' ) {
-		return ! isActive && isComplete
-			? translate( '%(googleMailService)s account information', {
-					args: {
-						googleMailService: getGoogleMailServiceFamily(),
-					},
-					comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
-			  } )
-			: translate( 'Enter your %(googleMailService)s account information', {
-					args: {
-						googleMailService: getGoogleMailServiceFamily(),
-					},
-					comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
-			  } );
+		return (
+			<>
+				{ ! isActive && isComplete
+					? String(
+							translate( '%(googleMailService)s account information', {
+								args: {
+									googleMailService: getGoogleMailServiceFamily(),
+								},
+								comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
+							} )
+					  )
+					: String(
+							translate( 'Enter your %(googleMailService)s account information', {
+								args: {
+									googleMailService: getGoogleMailServiceFamily(),
+								},
+								comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
+							} )
+					  ) }
+			</>
+		);
 	}
 
-	return ! isActive && isComplete
-		? translate( 'Billing information' )
-		: translate( 'Enter your billing information' );
+	return (
+		<>
+			{ ! isActive && isComplete
+				? String( translate( 'Billing information' ) )
+				: String( translate( 'Enter your billing information' ) ) }
+		</>
+	);
 };
 
 const OrderReviewTitle = () => {
 	const translate = useTranslate();
-	return translate( 'Your order' );
+	return <>{ String( translate( 'Your order' ) ) }</>;
 };
 
 const paymentMethodStep = getDefaultPaymentMethodStep();
 
 export default function WPCheckout( {
 	removeProductFromCart,
-	updateLocation,
-	applyCoupon,
-	removeCoupon,
-	couponStatus,
 	changePlanLength,
 	siteId,
 	siteUrl,
 	countriesList,
-	StateSelect,
 	getItemVariants,
-	responseCart,
```

Comment:
This can be pulled directly out of `useShoppingCart` so there's no need to pass it in.